### PR TITLE
fluent-gtk-theme: 2023-06-20 -> 2024-04-28

### LINKED ISSUE #307512: Update request: fluent-gtk-theme 2023-06-20 → 2024-04-28
- Package name: fluent-gtk-theme
- Latest released version: 2024-04-28

- Current version on the unstable channel: 2023-06-20
- Current version on the stable/release channel: 2023-06-20

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@romildo 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by romildo**
## Description of changes

Update to version [2024-04-28](https://github.com/vinceliuice/Fluent-gtk-theme/releases/tag/2024-04-28)

Closes https://github.com/NixOS/nixpkgs/issues/307512

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/themes/fluent-gtk-theme/default.nix [hunk 1]
```diff
@@ -23,13 +23,13 @@ lib.checkListOfEnum "${pname}: tweaks" [ "solid" "float" "round" "blur" "noborde
 
 stdenvNoCC.mkDerivation rec {
   inherit pname;
-  version = "2023-06-20";
+  version = "2024-04-28";
 
   src = fetchFromGitHub {
     owner = "vinceliuice";
     repo = pname;
```
Comment (JohnRTitor) [inline pkgs/data/themes/fluent-gtk-theme/default.nix:30]:
```suggestion
    repo = "fluent-gtk-theme";
```

File: pkgs/data/themes/fluent-gtk-theme/default.nix [hunk 2]
```diff
@@ -23,13 +23,13 @@ lib.checkListOfEnum "${pname}: tweaks" [ "solid" "float" "round" "blur" "noborde
 
 stdenvNoCC.mkDerivation rec {
```
Comment (JohnRTitor) [inline pkgs/data/themes/fluent-gtk-theme/default.nix:24]: Can you remove the recursion and move to `finalAttrs`?

File: pkgs/data/themes/fluent-gtk-theme/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/data/themes/fluent-gtk-theme/default.nix]: Done.

File: pkgs/data/themes/fluent-gtk-theme/default.nix (hunk 2 above)
Comment (romildo) [reply pkgs/data/themes/fluent-gtk-theme/default.nix]: Done.

File: pkgs/data/themes/fluent-gtk-theme/default.nix [hunk 3]
```diff
@@ -74,4 +74,4 @@ stdenvNoCC.mkDerivation rec {
     platforms = platforms.unix;
     maintainers = [ maintainers.romildo ];
```
Comment (JohnRTitor) [inline pkgs/data/themes/fluent-gtk-theme/default.nix:75]: Remove `with lib;` and use the components explicitly when needed.
For this particular line:
```suggestion
    maintainers = with lib.maintainers; [ romildo ];
```

The rest can be prepended with `lib.`

Comment (romildo) [reply pkgs/data/themes/fluent-gtk-theme/default.nix]: Done.

**Review by JohnRTitor — APPROVED** [review]

**Pull request merged**